scripts/named_entity_recognition/code/train_model.py

```diff
@@ -0,0 +1,379 @@
+        os.makedirs(os.path.split(log_path)[0])
+
+    # Configuring logger
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    fhandler = logging.FileHandler(log_path, mode='w')
+    shandler = logging.StreamHandler()
+    formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+    fhandler.setFormatter(formatter)
+    shandler.setFormatter(formatter)
+    logger.addHandler(fhandler)
+    logger.addHandler(shandler)
+
+    # prepare data
+    char_vocab, word_vocab, tag_vocab, \
+        train_dataloader, valid_dataloader, test_dataloader = data_pipeline(train_data_path, valid_data_path,
+                                                                            test_data_path,
+                                                                            embedding,
+                                                                            word_vocab_path,
+                                                                            char_vocab_path,
+                                                                            tag_vocab_path,
+                                                                            batch_size, char_len_per_word,
+                                                                            logger)
+
+    # build model
+    nchars, nwords, tag2idx = len(char_vocab), len(word_vocab), tag_vocab._token_to_idx
+    model = build_model(nchars, nchar_embed, nwords, nword_embed, nfilters, kernel_size,
+                        nhiddens, nlayers, ntag_space, emb_drop_prob, rnn_drop_prob, out_drop_prob,
+                        tag2idx,)
+
+    # use pretrained embedding
+    if embedding != 'random':
+        # print('setting embedding for layer')
+        model.word_embedding_layer.weight.set_data(word_vocab.embedding.idx_to_vec)
+        # fix or finetune
+        # model.word_embedding_layer.collect_params().setattr('grad_req', 'null')
+
+    # train and evaluate
+    train_and_valid(model, optim_name, train_dataloader, valid_dataloader, test_dataloader, nepochs,
+                    lr, clip, word_vocab, char_vocab, tag_vocab, lr_decay_step, lr_decay_rate, logger)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser('train model')
+
+    parser.add_argument('--train', dest='train_data_path', action='store')
+    parser.add_argument('--valid', dest='valid_data_path', action='store')
+    parser.add_argument('--test', dest='test_data_path', action='store')
+    parser.add_argument('--wvp', dest='word_vocab_path', action='store')
+    parser.add_argument('--cvp', dest='char_vocab_path', action='store')
+    parser.add_argument('--tvp', dest='tag_vocab_path', action='store')
+
+    parser.add_argument('--embedding', choices=['glove', 'senna', 'random'],
+                        help='Embedding for words', required=True)
+    parser.add_argument('--clpw', type=int, dest='char_len_per_word', default=12, action='store')
```

Comment:
I don't like the idea of using abbreviations in command line options, because that makes the shell script not readable without looking at the actual python code. Are we doing this in other scripts in the repo? I would suggest just using `--char-len-per-word` without `dest` and `store`.

Reply:
Ok, I thought about this when I was writing the code, but I don't know which option is better. Obviously, it seems that I have chosen a not so good way. I will modify him.

Reply:
I agree with yun on this, I think it makes sense to have --char-len-per-word kind of easily understandable notations.